cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -596,6 +598,54 @@ func (ht *hcsTask) DumpGuestStacks(ctx context.Context) string {
 	return ""
 }
 
+func (ht *hcsTask) Share(ctx context.Context, req *shimdiag.ShareRequest) error {
+	if ht.host == nil {
+		return errors.New("task is not isolated")
+	}
+	guestReq := guestrequest.GuestRequest{
+		ResourceType: guestrequest.ResourceTypeMappedDirectory,
+		RequestType:  requesttype.Add,
+	}
+	// If the task is WCOW then we'll be mounting the directory/file with VSMB
+	// and returning.
+	if ht.isWCOW {
+		options := &hcsschema.VirtualSmbShareOptions{
+			NoDirectmap: ht.host.DevicesPhysicallyBacked(),
+		}
+		if req.ReadOnly {
+			options.ReadOnly = true
+			options.CacheIo = true
+			options.ShareRead = true
+			options.ForceLevelIIOplocks = true
+		}
+		guestReq.Settings = &hcsschema.MappedDirectory{
+			ContainerPath: req.UvmPath,
+			HostPathType:  hcsschema.VirtualSmbShareName,
```

Comment:
Is there an advantage to using `HostPathType` and the share name instead of just using the path to the share (as we do for the mappings we set up in hcsdoc_wcow.go)?

Reply:
This was just modeled after hcsdiag is why I have it this way. Let me try supplying the full vsmb path instead of just the share name

Reply:
That works also, so I'll switch to that if we'd prefer

Reply:
I don't think I have a preference, was just curious why we chose this way here. It might be good to understand if there is reason to prefer one over the other.

Reply:
I'll find out if theres any difference between the two and get back to this

Reply:
Talked to Rafael. The HostPathType is used internally to construct the full VSMB path so from HCS' point of view there is no difference at all between the two ways of doing it. He recommends whichever way is easier. He did mention that HostPathType isnt available on all releases but its present in RS5+ so I think we'll just go with the way we do it in hcsdoc_wcow, seems the clearer/easier route.

Reply:
Sounds good to me